### ISSUE TITLE
Enhancement: [no-empty-function] Allow empty implemented interface methods

### ISSUE THREAD

**Issue by qstyler** [typescript-eslint/typescript-eslint#5897]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/no-empty-function/

### Description

There's a way to allow empty overridden methods when you extend a class, but not when you implement an interface.

### Fail

```typescript
export interface MigrationInterface {
    up(queryRunner: QueryRunner): Promise<any>;
    down(queryRunner: QueryRunner): Promise<any>;
}
```

### Pass

```typescript
export class First1666856520661 implements MigrationInterface {
  public async up(queryRunner: QueryRunner): Promise<void> {
    await queryRunner.query('SELECT SCHEMA_NAME() AS "schema_name"');
  }
  public async down(_queryRunner: QueryRunner): Promise<void> {}
}
```

**Comment by bradzacher** [comment]
Duplicate of #3841
